WikiWatershed/model-my-watershed
# Custom Weather: Add fields to Project model

## Overview

For each GWLF-E project, the user can now possibly upload Custom Weather Data. This will likely be uploaded to S3 (see #3274). The Project model is updated to track the key to the upload. There is also a field to set whether or not to use the custom weather data, in case the user wants to switch back to the default weather data without losing that which they've already uploaded.

The exact nature / specification of the key will be specified as an answer to #3274, but 511 characters _should be_ enough to contain it.

Also adds migrations.

Connects #3269 

### Demo

```sql
SELECT id,
       name,
       LEFT(ST_AsGeoJSON(area_of_interest), 64) AS area_of_interest,
       is_private,
       model_package,
       created_at,
       modified_at,
       user_id,
       is_activity,
       area_of_interest_name,
       LEFT(gis_data, 64) AS gis_data,
       wkaoi,
       mapshed_job_uuid_id,
       subbasin_mapshed_job_uuid_id,
       custom_weather_dataset,
       uses_custom_weather
FROM modeling_project
LIMIT 1;

-[ RECORD 1 ]----------------+-----------------------------------------------------------------
id                           | 2
name                         | Subbasin Legend
area_of_interest             | {"type":"MultiPolygon","coordinates":[[[[-75.638687083659,
is_private                   | t
model_package                | gwlfe
created_at                   | 2019-08-28 17:57:19.915671+00
modified_at                  | 2019-08-28 18:34:36.025278+00
user_id                      | 2
is_activity                  | f
area_of_interest_name        | Lower Schuylkill River, HUC-10 Watershed (ID 0204020310)
gis_data                     | {"n38": 0, "n39": 0, "n30": 0, "n31": 0, "n32": 0, "n33": 0,
wkaoi                        | huc10__1341
mapshed_job_uuid_id          | 81ecf121-229d-4544-9480-2aa12d968a4d
subbasin_mapshed_job_uuid_id | 0b2b3a38-bb9c-4818-a528-a80fe983eee4
custom_weather_dataset       |
uses_custom_weather          | f
```

## Testing Instructions

* Check out this branch
* Run migrations `./scripts/manage.sh migrate`
  - [ ] Ensure they run successfully
